client/components/fields/preview/SimpleList/Calendars.tsx

```diff
@@ -1,18 +1,33 @@
 import * as React from 'react';
+import {connect} from 'react-redux';
 import {get} from 'lodash';
 
 import {superdeskApi} from '../../../../superdeskApi';
-import {IListFieldProps} from '../../../../interfaces';
+import {ICalendar, IListFieldProps} from '../../../../interfaces';
+import {calendars as getCalendars} from '../../../../selectors/events';
 
 import {PreviewSimpleListItem} from './PreviewSimpleListItem';
+import {getVocabularyItemNames} from '../../../../utils/vocabularies';
 
-export class PreviewFieldCalendars extends React.PureComponent<IListFieldProps> {
+interface IProps extends IListFieldProps {
+    calendars: Array<ICalendar>;
+}
+
+const mapStateToProps = (state) => ({
+    calendars: getCalendars(state),
+});
+
+class PreviewFieldCalendarsComponent extends React.PureComponent<IProps> {
     render() {
         const {gettext} = superdeskApi.localization;
         const field = this.props.field ?? 'calendars';
-        const calendarNames = (get(this.props.item, field) || [])
-            .map((calendar) => calendar.name)
-            .join(', ');
+        const calendarNames = getVocabularyItemNames(
+            get(this.props.item, field) || [],
+            this.props.calendars,
+            'qcode',
+            'name',
+            this.props.language
+        );
```

Comment:
seems like the usage is pretty much the same, so maybe there could be some higher level helper to avoid some duplication